### PR TITLE
Enable Teams & Projects on joining of a team

### PR DESCRIPTION
**Pull request by AlexTugarev**
Closes https://github.com/gitpod-io/gitpod/issues/5476

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **alextugarev** after the PR has been reviewed.

Associated issue: *#5476*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/projects/NewProject.tsx
Comment (jankeromnes) [inline components/dashboard/src/projects/NewProject.tsx:55]: You could simplify this Effect's 🍝 code a bit by adding `if (!user) { return; }` at the top. (Also the empty line is unnecessary. 😇)

File: components/server/src/workspace/gitpod-server-impl.ts
Comment (jankeromnes) [inline components/server/src/workspace/gitpod-server-impl.ts:1444]: Optional: The common pattern seems to be getting a user via `this.check{,AndBlock}User()` and then using it as a variable. You could potentially pass `user` as an argument to `ensureTeamsEnabled`, and avoid the direct `this.user` access.

**Review by jankeromnes — APPROVED** [review]
Looks good to me! And does the job 👍 thanks @AlexTugarev.

Left two suggestions in-line. Please feel free to address or dismiss them. Then, please feel free to manually carry over the `lgtm` label, and to remove the hold (without re-requesting a review 🌴).

/approve
/lgtm
/hold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: db9b88b1308072ce3bbf5eeffa72612d48b35518</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5477#pullrequestreview-743728431" title="LGTM">jankeromnes</a>*

Associated issue: *#5476*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jankeromnes]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by gtsiolis — APPROVED** [review]
Works as expected! Thanks for adding this @AlexTugarev! 🧙 

For a UX perspective there's a small but barely noticeable change in the user interface when joining a team but this is probably a broader issue with the frontend app, right? See screen recording and notice the top menu changing two times and page error (`undefined`) before loading the team.


https://user-images.githubusercontent.com/120486/131655663-d812b0d3-6d0e-485a-a6d3-f8103e82224c.mov

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5477#pullrequestreview-743746515" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5477#pullrequestreview-743728431" title="LGTM">jankeromnes</a>*

Associated issue: *#5476*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jankeromnes]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/src/workspace/gitpod-server-impl.ts
Comment (AlexTugarev) [reply components/server/src/workspace/gitpod-server-impl.ts:1444]: the purpose of `ensureTeamsEnabled` is to modify the state of `this.user`, therefore this is somewhat exceptional.

Comment (jankeromnes) [reply components/server/src/workspace/gitpod-server-impl.ts:1444]: Makes sense, thanks. Also it should be removed again soon, so resolving this. 😊

File: components/dashboard/src/projects/NewProject.tsx
Comment (AlexTugarev) [reply components/dashboard/src/projects/NewProject.tsx:55]: I'd love to see that PR reverted rather soon, i.e. getting to general availability of Teams & Projects. I would be great to just revert this PR by then with less effort. WDYT?

**Comment by AlexTugarev** [comment]
wrt https://github.com/gitpod-io/gitpod/pull/5477#pullrequestreview-743746515

@gtsiolis, yes it seems to be related to the decoupled states of pages and menu. as we want the menu to render quickly, we cannot await the additional roundtrip.

**Comment by gtsiolis** [comment]
> yes it seems to be related to the decoupled states of pages and menu. as we want the menu to render quickly, we cannot await the additional roundtrip.

Thanks for clarifying @AlexTugarev!

**Pull request merged**
